src/base/preferences.cpp

```diff
@@ -1054,6 +1056,87 @@ void Preferences::setExecutionLogEnabled(bool b)
     setValue("Preferences/ExecutionLog/enabled", b);
 }
 
+int Preferences::executionLogMessageTypes() const
+{
+    // as default value we need all the bits set
+    // -1 is considered the portable way to achieve that
+    return value("Preferences/ExecutionLog/MessageTypes", -1).toInt();
+}
+
+void Preferences::setExecutionLogMessageTypes(const int &value)
+{
+    setValue("Preferences/ExecutionLog/MessageTypes", value);
+}
+
+// File log
+bool Preferences::fileLogEnabled() const
+{
+    return value("Preferences/FileLog/enabled", true).toBool();
+}
+
+void Preferences::setFileLogEnabled(bool enabled)
+{
+    setValue("Preferences/FileLog/enabled", enabled);
+}
+
+QString Preferences::fileLogPath() const
+{
+    return value("Preferences/FileLog/Path", QVariant(Utils::Fs::QDesktopServicesDataLocation() + LOG_FOLDER)).toString();
+}
+
+void Preferences::setFileLogPath(const QString &path)
+{
+    setValue("Preferences/FileLog/Path", path);
+}
+
+bool Preferences::fileLogManaged() const
+{
+    return value("Preferences/FileLog/managed", true).toBool();
+}
+
+void Preferences::setFileLogManaged(bool managed)
+{
+    setValue("Preferences/FileLog/managed", managed);
+}
+
+int Preferences::fileLogMaxSize() const
+{
+    int val = value("Preferences/FileLog/MaxSize", 10).toInt();
+    if (val < 1)
+        return 1;
+    else if (val > 1000)
```

Comment:
I don't understand what do you mean.
We have three cases:
val <= 1: we should return 1
1 < val <= 1000: we should return val itself
val > 1000: we should return 1000
and we have three branches in code which corresponds to three cases above. What is redundant?